troubadix/helper/linguistic_exception_handler.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (C) 2021-2022 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import re
+from abc import ABC, abstractmethod
+from typing import List, Tuple, Union
+
+
+class LinguisticCheck(ABC):
+    @abstractmethod
+    def execute(self, file, content):
+        pass
+
+
+class FileCheck(LinguisticCheck):
+    def __init__(self, file: str) -> None:
+        self.file = file
+
+    def execute(self, file: str, content: str) -> bool:
+        return self.file in file
+
+
+class FilesCheck(LinguisticCheck):
+    def __init__(self, files: List[str]) -> None:
+        self.files = files
+
+    def execute(self, file: str, content: str):
+        return any([_file in file for _file in self.files])
```

Comment:
Can we change that to a generator too? Haven't tried that yet
```suggestion
        return any(_file in file for _file in self.files)
```

Reply:
Yes, we can. Already have it on my local branch